From 1d7b2467ec7786375ca09f60772a41edb27e3e02 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:48:11 -0400
Subject: [pci] Remove pci_bus_b() and use list_for_each_entry() directly

Message-id: <20140806034810.1438.73489.stgit@gir.stowe>
Patchwork-id: 87447
O-Subject: [RHEL7.1 PATCH 38/69] PCI: Remove pci_bus_b() and use list_for_each_entry() directly
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 94e6a9b93064b49024b8701d2d81fcb4a821fa09

commit 94e6a9b93064b49024b8701d2d81fcb4a821fa09
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Feb 13 21:14:03 2014 +0800

    PCI: Remove pci_bus_b() and use list_for_each_entry() directly

    Replace list_for_each() with list_for_each_entry(), which means we no
    longer need pci_bus_b() and can remove it.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/pci.c b/drivers/pci/pci.c
index abb8213..d5c9908 100644
--- a/drivers/pci/pci.c
+++ b/drivers/pci/pci.c
@@ -108,12 +108,12 @@ static bool pcie_ari_disabled;
  */
 unsigned char pci_bus_max_busnr(struct pci_bus* bus)
 {
- struct list_head *tmp;
+ struct pci_bus *tmp;
  unsigned char max, n;
 
  max = bus->busn_res.end;
- list_for_each(tmp, &bus->children) {
-  n = pci_bus_max_busnr(pci_bus_b(tmp));
+ list_for_each_entry(tmp, &bus->children, node) {
+  n = pci_bus_max_busnr(tmp);
   if(n > max)
    max = n;
  }
diff --git a/drivers/pci/search.c b/drivers/pci/search.c
index 8041d02..60da0b3 100644
--- a/drivers/pci/search.c
+++ b/drivers/pci/search.c
@@ -54,14 +54,14 @@ pci_find_upstream_pcie_bridge(struct pci_dev *pdev)
 
 static struct pci_bus *pci_do_find_bus(struct pci_bus *bus, unsigned char busnr)
 {
- struct pci_bus* child;
- struct list_head *tmp;
+ struct pci_bus *child;
+ struct pci_bus *tmp;
 
  if(bus->number == busnr)
   return bus;
 
- list_for_each(tmp, &bus->children) {
-  child = pci_do_find_bus(pci_bus_b(tmp), busnr);
+ list_for_each_entry(tmp, &bus->children, node) {
+  child = pci_do_find_bus(tmp, busnr);
   if(child)
    return child;
  }
@@ -111,7 +111,7 @@ pci_find_next_bus(const struct pci_bus *from)
  down_read(&pci_bus_sem);
  n = from ? from->node.next : pci_root_buses.next;
  if (n != &pci_root_buses)
-  b = pci_bus_b(n);
+  b = list_entry(n, struct pci_bus, node);
  up_read(&pci_bus_sem);
  return b;
 }
diff --git a/include/linux/pci.h b/include/linux/pci.h
index 614d5e6..ad16945 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -492,7 +492,6 @@ struct pci_bus_rh {
 #endif
 };
 
-#define pci_bus_b(n) list_entry(n, struct pci_bus, node)
 #define to_pci_bus(n) container_of(n, struct pci_bus, dev)
 
 /*
-- 
1.7.1